draft-oauth-ai-agents.md

```diff
@@ -0,0 +1,356 @@
+
+The sub_parent and client_parent claims are used to represent the owning or parent application of the subject (sub) and client (client_id), respectively. These claims help link an agent instance to the broader application or deployment it belongs to, supporting policy decisions and traceability. They SHOULD be included when the subject or client is an agent instance to enable clear relationships between agents and their controlling applications.
+
+Authorization Servers MAY include the act (Actor) claim to explicitly represent delegation chains. To control token size and ensure predictable processing time at Resource Servers, Authorization Servers MAY limit the depth of nested act structures.
+
+act (Actor):
+: RECOMMENDED. Identifies the Agent currently performing the action (typically the same as the client_id). This claim is used to represent delegation chains where Agents act on behalf of others using Token Exchange ({{RFC8693}}).
+
+Authorization Servers SHOULD impose practical limits on the nesting depth of the act claim to prevent token size bloat and ensure efficient validation by Resource Servers. While no fixed limit is mandated, a typical recommendation is to restrict the chain to 3–5 delegation levels, balancing expressiveness and performance.
+
+## Example decoded JWT payload
+
+~~~
+{ 
+  "sub": "user-id-123", 
+  "sub_entity_type": "user", 
+  "aud": "https://api.example.com", 
+  "scope": "read:email write:calendar", 
+  "client_id": "agent-xyz-instance-id-456", 
+  "client_entity_type": "agent", 
+  "client_parent": "agent-xyz-app-789", 
+  "act": { 
+    "sub": "agent-xyz-instance-id-456", 
+    "sub_entity_type": "agent", 
+    "sub_parent": "agent-xyz-app-789", 
+    "act": { 
+      "sub": "agent-abc-instance-id-123", 
+      "sub_entity_type": "agent", 
+      "sub_parent": "agent-abc-app-1610" 
+    } 
+  } 
+}
+~~~
+
+# Authorization Server Behavior  
+
+Authorization Servers MUST:
+
+* Verify the Agent’s identity and permissions prior to issuing access tokens.
+* Confirm that the Agent is registered and authorized to act on behalf of the User or another Agent.
+* Include all required claims as specified in Section 4.
+* Provide mechanisms for users to review, manage, and revoke Agent access and consent.
+
+It is RECOMMENDED that Authorization Servers log all Agent-related actions and access events to support auditing and accountability.
+
+Authorization Servers MAY implement additional security controls such as rate limiting, IP allowlisting, or other measures to mitigate the risk of Agent misuse.
+
+## User Consent Experience
+
+Authorization Servers SHOULD clearly communicate to end-users the identity of the acting agent, the permissions requested, and the implications of consent (where applicable) during authorization prompts. Consent screens should be designed to be understandable, minimizing ambiguity about which agent is requesting access and what actions it may perform on the user’s behalf.
```

Comment:
Consider language to emphasisethat this ccoonsent may be obtained out-of-context and precautions should be taken against consent phishing attacks (ensuing the consent request is originating from a legitimate agent, and that there is enough information for the user to understand what they are consenting to.

In terms of terminology, it has been pointed out to me in the past that "consent"" is a legal term, what we are really doing in the scenarios is collecting permission. Tamoto/Tamatoe, but other reviewers will trip up on it.